src/hardware/serialport/softmodem.h

```diff
@@ -231,6 +231,9 @@ class CSerialModem final : public CSerial {
 	                              // false: send text (i.e. NO DIALTONE)
 	bool telnet_mode = false; // Default to direct null modem connection;
 	                         // Telnet mode interprets IAC
+	char connect_string[14];	// will hold the string to report on
+					// connect with specified or default
+					// speed
```

Comment:
Suggest using literal-sizing, and make the literal the maximum possible string length:

``` c++
char connect_string[] = "CONNECT 57600";
```

Reply:
This is a problem in the class initializer since it can also be set in the constructor.  I think the choice here is to specify the actual maximum length, unless you have an alternative proposal.

Reply:
Ahh, indeed. Let's go with:

``` c++
char connect_string[14] = {"CONNECT 57600"};
```

This does narrowing testing, so if the right-hand-side is too big, the compiler will give us an error. For example:

```
../../src/dos/../hardware/serialport/softmodem.h:234:29: error: initializer-string for char array is too long
        char connect_string[10] = {"CONNECT 57600"};
```

The content can still be set by the constructors (this is just the default initializer).